### PR TITLE
RFC: Add testing mechanism

### PR DESCRIPTION
**Pull request by 72636c**
I will write a changeset and tests if peeps are on board with this :)

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+[destinations]: https://github.com/pinojs/pino/blob/v9.2.1/docs/api.md#destination
+
+In practice, this may look like the following:
+
+```typescript
+import createLogger, { createDestination } from '@seek/logger';
+
+const { destination, stdoutMock } = createDestination({
+  mock: config.environment === 'test',
+});
```
Comment (72636c) [inline docs/testing.md:72]: Fielding opinions on this interface and the `stdoutMock` name in particular

File: src/destination/mock.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+     *
+     * Throws an error if more or less than 1 call has been recorded.
+     *
+     * This may be asserted against in a test case.
+     *
+     * ```typescript
+     * expect(stdoutMock.onlyCall()).toMatchSnapshot();
+     * ```
+     */
+    onlyCall: (): object => {
```
Comment (72636c) [inline src/destination/mock.ts:91]: Can we think of any other convenience methods worth implementing in a first cut?

File: src/destination/create.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * Returns:
+ *
+ * - A `destination` to be supplied to the second parameter of `createLogger()`
+ * - A `stdoutMock` recorder for testing and troubleshooting logging calls
+ */
+export const createDestination = Object.assign(
+  (opts: CreateDestinationOptions) => {
+    const stdoutMock = createStdoutMock(
+      typeof opts.mock === 'object' ? opts.mock : DEFAULT_MOCK_OPTIONS,
+    );
```
Comment (72636c) [inline src/destination/create.ts:34]: There is inherent overhead here in us initialising a `stdoutMock` even in production where `opts.mock: false`. However this should be truly negligible overhead that is only occurred once on startup.

**Review by AaronMoat — APPROVED** [review]
I really like this. Asserting log output is something that I don't do often but when I want to, it's been a giant pain.

**Review by AaronMoat — APPROVED** [review]
🙌

**Pull request merged**